_ZadanieDomowe_8_PokerHands_Base/PokerHands/Services/PokerHandEvaluator.cs

```diff
@@ -0,0 +1,73 @@
+﻿using PokerHands.Services.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PokerHands
+{
+    public class PokerHandEvaluator : IHandEvaluator
+    {
+        private readonly IHandService _handService;
+        private readonly ICardParser _cardParser;
+
+        public PokerHandEvaluator(IHandService handService, ICardParser cardParser)
+        {
+            _handService = handService;
+            _cardParser = cardParser;
+        }
+
+        public Combination WhatIsTheHighestCombination(int gameNumber, params string[] cards)
+        {
+            var parsedCards = cards.Select(card => _cardParser.ParseCardString(card));
+
+            var cardsfromLowest = _handService.GroupCardsFromLowest(parsedCards);
+            var cardGroupsByValues = _handService.GroupCardsByValues(cardsfromLowest).ToList();
+            var cardGroupWithMostOccurences = _handService.GroupByMostOccurences(cardGroupsByValues);
+
+            bool cardsAreSameColor = IsSameColor(cardsfromLowest);
+            bool isStraight = IsStraight(cardsfromLowest);
+
+            if (isStraight)
+            {
+                if (cardsAreSameColor)
+                {
+                    bool highestIsAce = cardsfromLowest.Last().Value == Value.Ace;
+                    return highestIsAce ? Combination.RoyalFlush : Combination.StraightFlush;
+                }
+                return Combination.Straight;
+            }
+            else if (cardsAreSameColor)
+            {
+                return Combination.Flush;
+            }
+            else if (cardGroupsByValues.Any(IsGroupOf(2)) && cardGroupsByValues.Any(IsGroupOf(3)))
```

Comment:
bardzo dobrze to wygląda. Tylko te słowa else są niepotrzebne ze względu na returny.